### PR TITLE
[KYUUBI #1528] Record the kyuubi server ip address in event log

### PR DESCRIPTION
**Pull request by wForget**
### _Why are the changes needed?_

Record the kyuubi server ip address in event log. For details: #1528

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [X] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala [hunk 1]
```diff
@@ -53,6 +54,7 @@ class SparkSessionImpl(
       case (key, value) => setModifiableConfig(key, value)
     }
     KDFRegistry.registerAll(spark)
+    sessionEvent.serverIp = serverIpAddress
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala:57]: inside `SessionEvent.apply`

**Comment by ulysses-you** [comment]
since this changes UI, can you provide the screen snapshot ?

**Review by yaooqinn — APPROVED** [review]

**Comment by wForget** [comment]
> since this changes UI, can you provide the screen snapshot ?

I recorded the screenshots in the issue #1528.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala (hunk 1 above)
Comment (wForget) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala:57]: > inside `SessionEvent.apply`

change `SessionEvent.apply(session: Session)` to `SessionEvent.apply(session: SparkSessionImpl)`, or use `session.asInstanceOf[SparkSessionImpl].serverIpAddress` in SessionEvent, which one is better?

Comment (ulysses-you) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala:57]: I prefer the first option.

Comment (wForget) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala]: > I prefer the first option.

Thanks @ulysses-you , I changed it to that.

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**